src/resolver.cpp

```diff
@@ -128,7 +128,16 @@ std::string resolveToPath(const openassetio::hostApi::ManagerPtr &manager,
   static constexpr std::string_view kFileURLScheme{"file://"};
   static constexpr std::size_t kProtocolSize = kFileURLScheme.size();
 
-  openassetio::Str url = LocatableContentTrait(traitsData).getLocation();
+  auto urlOpt = LocatableContentTrait(traitsData).getLocation();
+
+  if (!urlOpt.has_value()) {
+    std::string msg = "Location not found in resolved entityReference: ";
+    msg += entityReference.toString();
+    throw std::runtime_error(msg);
+  }
+
+  auto url = urlOpt.value();
```

Comment:
[consider][minor] You already know it has a value, so can `*urlOpt` (though sometimes clang-tidy is too stupid to notice).

Reply:
https://github.com/OpenAssetIO/usdOpenAssetIOResolver/pull/38/commits/2fc1f78f862a0b7462b7d66e09ee9b9c81318eed